app/src/ts.app.js

```diff
@@ -55,7 +58,7 @@ class Listener {
 				return this;
 			}
 			let content = JSON.parse(e.data.replace(BROADCAST_PREFIX, ''));
-			if (content.key !== key) {
+			if (content.key !== key || content.appIds !== this.appIds) {
```

Comment:
or if we really want to go crazy, have `this.appIds` be a type `string` or `array`